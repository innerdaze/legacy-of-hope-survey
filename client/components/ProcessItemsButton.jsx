import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Button from 'grommet/components/Button'

class ProcessItemsButton extends Component {
  constructor(props) {
    super(props)

    this.state = {
      canProcess: true
    }

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick(e) {
    e.preventDefault()
    this.props.onProcessItemsClick()
  }

  render() {
    return (
      <Button
        primary
        disabled={this.state.canProcess}
        label='Send'
        onClick={this.handleClick}
      />
    )
  }
}

ProcessItemsButton.propTypes = {
  onProcessItemsClick: PropTypes.func.isRequired
}

ProcessItemsButton.defaultProps = {
  onProcessItemsClick: Function.prototype
}

export default ProcessItemsButton
