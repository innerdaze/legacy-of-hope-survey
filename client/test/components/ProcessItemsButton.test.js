import React from 'react'
import ProcessItemsButton from '../../components/ProcessItemsButton'

let wrapper, onProcessItemsClick

describe('Testing ProcessItemsButton...', () => {
  beforeEach(() => {
    onProcessItemsClick = jest.fn()
    wrapper = shallow(
      <ProcessItemsButton onProcessItemsClick={onProcessItemsClick} />
    )
  })

  describe('ProcessItemsButton layout', () => {
    test('Expect ProcessItemsButton to have a Button', () => {
      expect(wrapper.find('Button')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect onProcessItemsClick is called when Button click', () => {
      wrapper.find('Button').simulate('click', {
        preventDefault() {}
      })

      expect(onProcessItemsClick).toBeCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on state', () => {
    test('Expect default state of canProcess is true', () => {
      expect(wrapper.state('canProcess')).toEqual(true)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
