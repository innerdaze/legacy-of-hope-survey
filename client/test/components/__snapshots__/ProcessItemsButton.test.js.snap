// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ProcessItemsButton... ProcessItemsButton layout Expect ProcessItemsButton to have a Button 1`] = `
<Button
  disabled={true}
  label="Send"
  method="push"
  onClick={[Function]}
  primary={true}
  type="button"
/>
`;

exports[`Testing ProcessItemsButton... Test on prop Expect onProcessItemsClick is called when Button click 1`] = `
<Button
  disabled={true}
  label="Send"
  method="push"
  onClick={[Function]}
  primary={true}
  type="button"
/>
`;

exports[`Testing ProcessItemsButton... Test on state Expect default state of canProcess is true 1`] = `
<Button
  disabled={true}
  label="Send"
  method="push"
  onClick={[Function]}
  primary={true}
  type="button"
/>
`;
